playbooks/itsm_inventory.yaml

```diff
@@ -0,0 +1,171 @@
+#           }] }}"
+#       # no_log: true
+#       loop: "{{ ansible_facts.packages | dict2items() }}"
+#       loop_control:
+#         loop_var: package
+#         label: "{{ package.key }}"
+#       when: >
+#         package.value[0].name | regex_search("^[l]")
+
+
+      - name: Inventory Software [m-z]
+        ansible.builtin.set_fact:
+         cacheable: false
+         software: "{{ software | default([]) + [{
+             'name': package.value[0].name,
+             'category': package.value[0].category | default(''),
+             'version': package.value[0].version
+           }] }}"
+       # no_log: true
+        loop: "{{ ansible_facts.packages | dict2items() }}"
+        loop_control:
+          loop_var: package
+          label: "{{ package.key }}"
+        when: >
+          package.value[0].name | regex_search("^[m-z]")
+
+
+      - name: Inventory Document
+        ansible.builtin.set_fact:
+          report: {
+            "details": "{{ details }}",
+            "os": {
+             "name": "{{ ansible_distribution | lower }}",
+             "version": "{{ ansible_distribution_version }}",
+             "version_major": "{{ ansible_distribution_major_version }}"
+            },
+            "software": "{{ software }}"
+          }
+
+
+      - name: Save report
+        ansible.builtin.copy:
+          content: "{{ report | to_nice_json }}"
+          dest: "/tmp/{{ ansible_hostname }}.json"
+
+    
+      - name: Upload inventory - {{ ansible_hostname }}
+        ansible.builtin.uri:
+          url: |-
+            {{ lookup('env', 'ITSM_API') }}/api/device/inventory
+
+          method: POST
+          body_format: json
+          src: "/tmp/{{ ansible_hostname }}.json"
+          remote_src: true
+          headers:
+            Authorization: Token {{ lookup('env', 'ITSM_TOKEN') }}
+          validate_certs: "{{ lookup('env', 'ITSM_VALIDATE_CERTS') | default(true) | bool }}"
+          timeout: 300
+          status_code:
+            - 200
+            - 201
+        no_log: >    # Contains a secret that logging shows
+          {{ nfc_pb_disable_log | default(true) }}
+  
+      always:
+        - name: Remove report
+          ansible.builtin.file:
+            path:  "/tmp/{{ ansible_hostname }}.json"
+            state: absent
+
+  vars:
+
+    nfc_pb_awx_tower_template:
+
+      - name: "ITSM/Machine/Inventory"
+        ask_tags_on_launch: false
+        ask_inventory_on_launch: true
+        ask_credential_on_launch: true
+        ask_limit_on_launch: true
+        concurrent_jobs_enabled: true
+        description: Collect inventory of host machines and publish to itsm django
+        execution_environment: "No Fuss Computing EE"
+        job_type: "run"
+        # job_tags: complete
+        labels:
+          - itsm
+          - itam
+          - inventory
+        use_fact_cache: true
+        credential_types:
+          - name: 'Playbook/Inventory/Django ITSM'
```

Comment:
- [ ] check this to ensure that it's using the current var names.
- [ ] update to reflect product name

current env vars are part of the inventory plugin

Reply:
updated in commit 26c849a6ebbad8d573d8fc18a49540d520062b3e